### PR TITLE
Rewrite book chapter8 machine translation documentation and train.py

### PR DESCRIPTION
**Pull request by nickyfantasy**

### CODE REVIEW

File: 08.machine_translation/README.md [hunk 1]
```diff
@@ -208,338 +164,323 @@ Because the full dataset is very big, to reduce the time for downloading the ful
...
+import numpy as np
+import paddle
+import paddle.fluid as fluid
+import paddle.fluid.framework as framework
+import paddle.fluid.layers as pd
+from paddle.fluid.executor import Executor
+from functools import partial
+import os
+
+dict_size = 30000
```
Comment (daming-lu) [inline 08.machine_translation/README.md:183]: Please add the old comments here : `# dict dim`

File: 08.machine_translation/README.md [hunk 2]
```diff
@@ -208,338 +164,323 @@ Because the full dataset is very big, to reduce the time for downloading the ful
...
 
    ```python
-   backward_first = paddle.layer.first_seq(input=src_backward)
-   decoder_boot = paddle.layer.fc(
-         size=decoder_size,
-         act=paddle.activation.Tanh(),
-         bias_attr=False,
-         input=backward_first)
+   def encoder(is_sparse):
+    # encoder
```
Comment (daming-lu) [inline 08.machine_translation/README.md:199]: we do not need this comment

File: 08.machine_translation/README.md [hunk 3]
```diff
@@ -208,338 +164,323 @@ Because the full dataset is very big, to reduce the time for downloading the ful
...
-            act=paddle.activation.Softmax(),
-            input=gru_step)
-        return out
-   ```
+Implement the decoder for training as follows:
 
-4. Define the name for the decoder and the first two input for `gru_decoder_with_attention`. Note that `StaticInput` is used for the two inputs. Please refer to [StaticInput Document](https://github.com/PaddlePaddle/Paddle/blob/develop/doc/v2/howto/rnn/recurrent_group_en.md#input) for more details.
+```python
+   def train_decoder(context, is_sparse):
+    # decoder
```
Comment (daming-lu) [inline 08.machine_translation/README.md:219]: nor this comment

File: 08.machine_translation/README.md [hunk 4]
```diff
@@ -208,338 +164,323 @@ Because the full dataset is very big, to reduce the time for downloading the ful
...
-            output_layer=beam_gen,
-            parameters=parameters,
-            input=gen_data,
-            field=['prob', 'id'])
-   ```
-4. Print generated translation
+### Define DataSet
 
-    Print sequence and its `beam_size` generated translation results based on the dictionary.
+We initialize ids and scores and create tensors for input. This test we are using first record data from `wmt14.test` for inference. At the end we get src dict and target dict for printing out results later.
```
Comment (daming-lu) [inline 08.machine_translation/README.md:424]: In this test, we are using the first record data ...

File: 08.machine_translation/README.md [hunk 5]
```diff
@@ -208,338 +164,323 @@ Because the full dataset is very big, to reduce the time for downloading the ful
...
-               end_pos = gen_sen_idx[i * beam_size + j]
-               print("%.4f\t%s" % (beam_result[0][i][j], " ".join(
-                     trg_dict[w] for w in beam_result[1][start_pos:end_pos])))
-               start_pos = end_pos + 2
-           print("\n")
-   ```
+```python
+init_ids_data = np.array([1 for _ in range(batch_size)], dtype='int64')
+init_scores_data = np.array(
+    [1. for _ in range(batch_size)], dtype='float32')
```
Comment (daming-lu) [inline 08.machine_translation/README.md:429]: they could simply do this `[1.0] * batch_size`

File: 08.machine_translation/train.py [hunk 6]
```diff
@@ -1,235 +1,273 @@
...
-            print("\n")
+import paddle
+import paddle.fluid as fluid
+import paddle.fluid.framework as framework
+import paddle.fluid.layers as pd
+from paddle.fluid.executor import Executor
+from functools import partial
+import os
+
+dict_size = 30000
```
Comment (daming-lu) [inline 08.machine_translation/train.py:25]: add the old comments here

**Review by daming-lu — APPROVED** [review]
need some fixes
